components/zcash_address/src/kind/unified.rs

```diff
@@ -332,47 +568,54 @@ pub(crate) mod private {
                     return Err(ParseError::InvalidTypecodeOrder);
                 } else if t_code == prev_code {
                     return Err(ParseError::DuplicateTypecode(t));
-                } else if t == Typecode::P2sh && prev_code == Some(u32::from(Typecode::P2pkh)) {
+                } else if t == Typecode::Data(DataTypecode::P2sh)
+                    && prev_code == Some(u32::from(DataTypecode::P2pkh))
+                {
                     // P2pkh and P2sh can only be in that order and next to each other,
                     // otherwise we would detect an out-of-order or duplicate typecode.
                     return Err(ParseError::BothP2phkAndP2sh);
                 } else {
                     prev_code = t_code;
-                    only_transparent = only_transparent && t.is_transparent();
+                    only_transparent = only_transparent && item.is_transparent_data_item();
                 }
             }
 
             if only_transparent {
                 Err(ParseError::OnlyTransparent)
             } else {
                 // All checks pass!
-                Ok(Self::from_inner(items))
+                Ok(Self::from_inner(revision, items))
             }
         }
 
         fn parse_internal<T: Into<Vec<u8>>>(hrp: &str, buf: T) -> Result<Self, ParseError> {
-            Self::parse_items(hrp, buf).and_then(Self::try_from_items_internal)
+            Self::parse_items(hrp, buf)
+                .and_then(|(revision, items)| Self::try_from_items_internal(revision, items))
         }
     }
 }
 
-use private::SealedItem;
+use private::SealedDataItem;
 
 /// Trait providing common encoding and decoding logic for Unified containers.
 pub trait Encoding: private::SealedContainer {
-    /// Constructs a value of a unified container type from a vector
-    /// of container items, sorted according to typecode as specified
-    /// in ZIP 316.
+    /// Constructs a value of a unified container type from a vector of container
+    /// items. These items will be sorted according to typecode as specified in ZIP
+    /// 316, so this method is not necessarily round-trip compatible with
+    /// [`Container::items_as_parsed`].
     ///
     /// This function will return an error in the case that the following ZIP 316
     /// invariants concerning the composition of a unified container are
     /// violated:
     /// * the item list may not contain two items having the same typecode
     /// * the item list may not contain only transparent items (or no items)
```

Comment:
I thought we had decided to drop the prohibition on an R1 UA/UVK containing only transparent items? The motivation for doing that holds even though we are doing TEX addresses, because it is possible for an R1 UA/UVK to include metadata for a transparent address that is inexpressible using the bare t-addr/viewing key formats.

Reply:
Ah, yeah, this doc is now wrong.